crates/uv/src/lib.rs

```diff
@@ -42,8 +44,59 @@ pub(crate) mod printer;
 pub(crate) mod settings;
 pub(crate) mod version;
 
+/// Resolves the script target for a run command.
+///
+/// If it's a local file, does nothing. If it's a URL, downloads the content
+/// to a temporary file and updates the command. Prioritizes local files over URLs.
+/// Returns Some(NamedTempFile) if a remote script was downloaded, None otherwise.
+async fn resolve_script_target(
+    command: &mut ExternalCommand,
+) -> Result<Option<tempfile::NamedTempFile>> {
+    let Some(target) = command.get_mut(0) else {
+        return Ok(None);
+    };
+
+    // Only continue if we are absolutely certain no local file exists.
+    //
+    // We don't do this check on Windows since the file path would
+    // be invalid anyway, and thus couldn't refer to a local file.
+    if cfg!(unix) && !matches!(Path::new(target).try_exists(), Ok(false)) {
+        return Ok(None);
+    }
+
+    let maybe_url = target.to_string_lossy();
+
+    // Only continue if the target truly looks like a URL.
+    if !(maybe_url.starts_with("http://") || maybe_url.starts_with("https://")) {
+        return Ok(None);
+    }
+
+    let script_url = url::Url::parse(&maybe_url)?;
+    let file_stem = script_url
+        .path_segments()
+        .and_then(std::iter::Iterator::last)
+        .and_then(|segment| segment.strip_suffix(".py"))
+        .unwrap_or("script");
+
+    let temp_file = tempfile::Builder::new()
+        .prefix(file_stem)
+        .suffix(".py")
+        .tempfile()?;
+
+    eprintln!(
+        "Downloading remote script to: {}",
+        temp_file.path().display().cyan()
+    );
```

Comment:
Hmmmm. This is an issue I think. We should be using a `Printer` here derived from config settings. Otherwise this message won't be squashed when, e.g., `--quiet` is passed. I'm not quite sure how to solve that. Another approach would be to emit this message after a `Printer` is constructed below. See:

```
    // Configure the `Printer`, which controls user-facing output in the CLI.
    let printer = if globals.quiet {
        Printer::Quiet
    } else if globals.verbose > 0 {
        Printer::Verbose
    } else if globals.no_progress {
        Printer::NoProgress
    } else {
        Printer::Default
    };
```

Reply:
I updated the script to check whether the tempfile exists and report that a file was downloaded (reusing the `Printer`).

I think this is the most minimal way to respect `--quiet`. From an end user perspective, it would probably be nicer to report when we are actually starting to download the script, but I'm not quite should how to solve that without a bigger refactor.

One idea would be to create the temp file immediately and kick off the download, but await a future for the download/writing after writing "Downloading remote script...". Maybe too complicated.

Reply:
I think what you have is fine for now. Thanks!